apps/prairielearn/assets/scripts/lib/questionsTable.tsx

```diff
@@ -233,6 +233,7 @@ onDocumentReady(() => {
       searchInputs.forEach((searchInput) => {
         searchInput.setAttribute(
           'aria-label',
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
           `Filter by ${searchInput.closest('th')?.querySelector('div.th-inner')?.textContent?.trim()}`,
```

Comment:
This case is a little strange, there appears to be a bug with `textContent` specifically.

Reply:
What's the bug here? AFAICT it should always be non-null for an `Element` type like that returned by `querySelector`.

Reply:
It seems to infer the type as `(property) Node.textContent: string | null` even with `searchInput.closest('th')?.querySelector<HTMLDivElement>('div.th-inner')?.textContent`

> If the node is a [document](https://developer.mozilla.org/en-US/docs/Web/API/Document) or a [doctype](https://developer.mozilla.org/en-US/docs/Glossary/Doctype), textContent returns [null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null).

Reply:
That's not what I see:

<img width="710" height="91" alt="Screenshot 2025-09-30 at 14 26 32" src="https://github.com/user-attachments/assets/66f66f92-a226-4d21-bc7c-704cf8066685" />

Reply:
Fixed in 80790a1e69af46f3f60f2049106c0ce51bde835e.

Reply:
Interesting, did you editor not pick up on this?

https://github.com/PrairieLearn/PrairieLearn/blob/2e4786d61f93c712480ff0f44492eb0d42edcddb/.vscode/settings.json#L31

Regardless, glad it's fixed 🙏